setup.sh

```diff
@@ -76,6 +79,13 @@ function check_dependencies()
     fi
     eval "sudo $cmd"
   fi
+
+  for package in "${pip_packages[@]}"; do
+    pip list | grep -F "$package" > /dev/null
+    [[ "$?" != 0 ]] && pip_package_list="$package $pip_package_list"
+  done
+  cmd="pip install $pip_package_list"
+  eval "$cmd"
```

Comment:
Shouldn't we ask the user before running this?

Reply:
Why?

Reply:
I mean, we ask the user before installing the other dependencies, why should this one be different?

Reply:
Before installing a system dependency, we ask if the user wants to proceed; we then install everything, including this pip package. Also, pip installation in the userspace sounds like a routine task that nobody cares about. I can make it optional if you want, but I don't see the value in doing that.

Reply:
Ok, how about we move the for loop to line 72-73 and add `pip_package_list`  to the `ask_yN` prompt? Also you need to check if `pip_package_list` is not zero, as running `pip install` without arguments produces an error.
I left a comment suggestion with what I mean.